jwcrypto/jwe.py

```diff
@@ -289,6 +290,56 @@ def unwrap(self, key, keylen, ek, headers):
         return cek
 
 
+class _Pbes2HsAesKw(_RawKeyMgmt):
+
+    def __init__(self, hashsize, keysize):
+        self.backend = default_backend()
+        self.hashsize = hashsize
+        self.keysize = keysize // 8
+
+    def _get_key(self, alg, key, p2s, p2c):
```

Comment:
`jwk = {"kty": "oct", "k": "thisismypassword"}`

If you accidentally pass that to, say, A128GCMKW, it will work; but be horribly insecure. I like the feature of PBES2 only accepting a string password and A128GCMKW only accepting a JWK. It provides type separation so that the key for one cannot be used in the other.